pkg/vm/engine/disttae/types.go

```diff
@@ -38,13 +41,26 @@ const (
 
 const (
 	// default database id for catalog
-	MO_CATALOG_ID  = 0
+	MO_CATALOG_ID  = 1
 	MO_DATABASE_ID = 1
 	MO_TABLES_ID   = 2
 	MO_COLUMNS_ID  = 3
 )
 
-type Reader interface {
+// tae's block metadata, which is currently just an empty one, does not serve any purpose
+// When tae submits a concrete structure, it will replace this structure with tae's code
+type BlockMeta struct {
+}
+
+// Cache is a multi-version cache for maintaining some table data
+// The cache is concurrently secure, with multiple transactions accessing the cache at the same time
+type Cache interface {
```

Comment:
If there are 4 DNs, let's say, and your query touches only one, you poll only one.     Where is the infomation for this?   In context.Context?